pom.xml

```diff
@@ -95,6 +95,9 @@
     <scmTag>HEAD</scmTag>
     <!-- Where to put temporary files during test runs. -->
     <surefireTempDir>${project.build.directory}/tmp</surefireTempDir>
+
+    <!-- Generate metadata for reflection on method parameters -->
+    <maven.compiler.parameters>true</maven.compiler.parameters>
```

Comment:
Is there a reason you did not do this in `<configuration>` of the plugin? Is there some expectation that some plugins would want to opt out?

Reply:
> Is there a reason you did not do this in `<configuration>` of the plugin?

Is there a reason I should have?

> Is there some expectation that some plugins would want to opt out?

No expectation of that, but I thought it best to provide an escape hatch should this cause some unexpected problem.

Reply:
> Is there a reason I should have?

Just generally better style to typed configuration that is explicitly scoped to a particular mojo, unless there is a specific expectation that the property will be overridden sometimes.

Reply:
Is there an action item for me here?

Reply:
Not really.

Reply:
OK, I'll take this as a note for future work then. I'm happy to redo this if you think there is a need, but right now it seems good enough.